### PR TITLE
Build out the rest of the Now methods

### PR DESCRIPTION
**Pull request by nekevss**
This PR implements the remaining methods for `Now`'s API.

### CODE REVIEW

File: src/components/now.rs [hunk 1]
```diff
@@ -10,35 +10,110 @@ use crate::{iso::IsoDateTime, TemporalUnwrap};
...
     calendar::Calendar,
     timezone::{TimeZone, TzProvider},
-    EpochNanoseconds, Instant, PlainDateTime,
+    EpochNanoseconds, Instant, PlainDate, PlainDateTime, PlainTime, ZonedDateTime,
 };
 
+#[cfg(feature = "experimental")]
+use crate::{components::timezone::TZ_PROVIDER, TemporalError};
+#[cfg(feature = "experimental")]
+use std::ops::Deref;
```
Comment (jedel1043) [inline src/components/now.rs:19]: Hmm, I think if using `deref` needs to be gated, it would be better to use the `*` syntax.

**Review by jedel1043 — COMMENTED** [review]
Looks overall great!! I just have a small suggestion.

File: src/components/now.rs (hunk 1 above)
Comment (nekevss) [reply src/components/now.rs]: Yeah, I wasn't the biggest fan of using it at the time. I'm fine removing it.

**Review by jedel1043 — APPROVED** [review]
Nice!

**Pull request merged**